pkg/controller/appdefinition/pullappimage.go

```diff
@@ -34,11 +34,21 @@ func PullAppImage(transport http.RoundTripper) router.HandlerFunc {
 			return nil
 		}
 
-		appImage, err := images.PullAppImage(req.Ctx, req.Client, appInstance.Namespace, resolvedImage, "", remote.WithTransport(transport))
+		remoteDigest, err := images.ImageDigest(req.Ctx, req.Client, appInstance.Namespace, resolvedImage, remote.WithTransport(transport))
 		if err != nil {
 			cond.Error(err)
 			return nil
 		}
+
+		appImage := &v1.AppImage{}
+		// No pull needed if the remote digest matches the current digest
+		if remoteDigest != appInstance.Status.AppImage.Digest {
+			appImage, err = images.PullAppImage(req.Ctx, req.Client, appInstance.Namespace, resolvedImage, "", remote.WithTransport(transport))
+			if err != nil {
+				cond.Error(err)
+				return nil
+			}
+		}
 		appImage.Name = targetImage
```

Comment:
Non-blocking:
If there is a pull, we'll have all image details here (digest, etc.) right away.
Can't we just keep the current details if there is no pull (since we're not really changing the image)?

Reply:
I agree with this. If I understand correctly, addressing this may be as simple as moving lines 52 and 55 inside the added if-statement.